### PR TITLE
[esp32] Add lighting example

### PR DESCRIPTION
**Pull request by chshu**
#### Problem
Somehow, a simple Lighting example is missing for ESP platform. The all-cluster-app is too complex as a lighting device reference.

#### Change overview
Add lighting example for ESP platform, also add the new ESP32-S3 platform support.

#### Testing
Verified on ESP32, ESP32-C3, ESP32-S3.

### CODE REVIEW

File: examples/lighting-app/esp32/main/DeviceCallbacks.cpp [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+    if (event->SessionEstablished.IsCommissioner)
+    {
+        ESP_LOGI(TAG, "Commissioner detected!");
+    }
+}
+
+void DeviceCallbacks::OnOnOffPostAttributeChangeCallback(EndpointId endpointId, AttributeId attributeId, uint8_t * value)
+{
+    VerifyOrExit(attributeId == ZCL_ON_OFF_ATTRIBUTE_ID, ESP_LOGI(TAG, "Unhandled Attribute ID: '0x%04x", attributeId));
+    VerifyOrExit(endpointId == 1 || endpointId == 2, ESP_LOGE(TAG, "Unexpected EndPoint ID: `0x%02x'", endpointId));
```
Comment (gjc13) [inline examples/lighting-app/esp32/main/DeviceCallbacks.cpp:155]: If endpoint 2 doesn't really have a handler, shall we consider removing it?

**Review by gjc13 — APPROVED** [review]

File: examples/lighting-app/esp32/main/DeviceCallbacks.cpp [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+
+void DeviceCallbacks::PostAttributeChangeCallback(EndpointId endpointId, ClusterId clusterId, AttributeId attributeId, uint8_t mask,
+                                                  uint8_t type, uint16_t size, uint8_t * value)
+{
+    ESP_LOGI(TAG, "PostAttributeChangeCallback - Cluster ID: '0x%04x', EndPoint ID: '0x%02x', Attribute ID: '0x%04x'", clusterId,
+             endpointId, attributeId);
+
+    switch (clusterId)
+    {
+    case ZCL_ON_OFF_CLUSTER_ID:
```
Comment (bzbarsky-apple) [inline examples/lighting-app/esp32/main/DeviceCallbacks.cpp:100]: Please use the namespaced cluster ids (app::Clusters:OnOff:Id, etc).  If you `using namespace chip::app::Clusters;` those can be quite short.

File: examples/lighting-app/esp32/main/DeviceCallbacks.cpp [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+{
+    if (event->SessionEstablished.IsCommissioner)
+    {
+        ESP_LOGI(TAG, "Commissioner detected!");
+    }
+}
+
+void DeviceCallbacks::OnOnOffPostAttributeChangeCallback(EndpointId endpointId, AttributeId attributeId, uint8_t * value)
+{
+    VerifyOrExit(attributeId == ZCL_ON_OFF_ATTRIBUTE_ID, ESP_LOGI(TAG, "Unhandled Attribute ID: '0x%04x", attributeId));
```
Comment (bzbarsky-apple) [inline examples/lighting-app/esp32/main/DeviceCallbacks.cpp:154]: Please use the namespaced attribute ids (`Clusters::OnOff::Attributes::OnOff::Id`, etc).  Again, sane use of `using` keeps the verbosity down.

**Review by bzbarsky-apple — APPROVED** [review]

**Review by andy31415 — CHANGES_REQUESTED** [review]
Please also add build_examples.py support for compilation.

**Review by andy31415 — APPROVED** [review]

**Pull request merged**